build.go

```diff
@@ -646,6 +647,11 @@ func (c *Client) processBuildpacks(ctx context.Context, builderImage imgutil.Ima
 			fetchedBPs = append(append(fetchedBPs, mainBP), depBPs...)
 			order = appendBuildpackToOrder(order, mainBP.Descriptor().Info)
 		case buildpack.RegistryLocator:
+			r := regexp.MustCompile(`^(https|git)(://|@)([^/:]+)[/:]([^/:]+)/(.+).git$`)
+			if r.MatchString(registry) {
+				return fetchedBPs, order, errors.New("registry should be only specified as named")
+			}
+
 			registryCache, err := c.getRegistry(c.logger, registry)
```

Comment:
There may be some confusion on the original requirements. The flag should no longer accept a URI at all and instead only support **name**s from the `config.toml` files.

This is taken from the [latest PR changes](https://github.com/buildpacks/pack/pull/716#issue-441624401):

> By default, if no configuration values are defined in the `config.toml` file, buildpacks will be registered against:
> `https://github.com/buildpacks/registry`. In addition, user can specify their own registries. For example:
> 
> ```
> [[registries]]
> name="elbandito"
> type="github"
> url="https://github.com/elbandito/buildpack-registry"
> 
> [[registries]]
> name="joe"
> type="github"
> url="https://github.com/jkutner/buildpack-registry/"
> ```
> 
> When there are multiple `registries`, the user can specify a specific registry via `buildpack-registry, -R` flag.
> 
> For example:
> `pack register-buildpack <buildpack/image> -r elbandito`

- See also https://github.com/buildpacks/pack/issues/745

A possible solution may be to change `c.getRegistry` to accept a "name" and then it can then look up the URL from the `config.toml` instead of it being passed in.